### DISCUSSION TITLE
Create backup image from USB / SDCARD. How to run run armbian-tf on next run

### DISCUSSION THREAD

**Discussion by araczkowski** [ophub/amlogic-s9xxx-armbian#1646] [Amlogic] [answered]
Hi,
I'm creating a backup image from USB/SD Card with my programs. I'm using Gparted to shrink the partition and have smaller image.
Now the problem is that when I run the system from the backup I want to automatically resize the partition to the whole space on SD card. This can be done via ``armbian-tf `` command. But the question is how to execute this command automatically when the new backup is started for the first time?
Thanks for the answer

**Comment by ophub** [comment]
echo "yes" >/root/.no_rootfs_resize



https://github.com/ophub/amlogic-s9xxx-armbian/blob/f0af99363de413908ad2cddeac1b512667477ac4/build-armbian/armbian-files/common-files/usr/sbin/armbian-fix#L207-L212

**Comment by araczkowski** [comment]
thanks @ophub I found this flag and I set it to ``yes``
```
echo "yes" >/root/.no_rootfs_resize

```

but after new boot the flag is still ``yes`` but the size not adjusted
![image](https://github.com/ophub/amlogic-s9xxx-armbian/assets/1263968/aa50c0c6-b68a-4410-ad1d-ee9bec15d95a)


there is a flag ``armbian_firstrun`` maybe this one should be set?

https://github.com/ophub/amlogic-s9xxx-armbian/blob/f0af99363de413908ad2cddeac1b512667477ac4/rebuild#L1004

**Comment by araczkowski** [comment]
I also tried 
```
touch /root/.not_logged_in_yet
```
but still no root partition resized

when the ``armbian-fix`` is executed?

**Comment by ophub** [comment]
When build armbian os add:


`echo "yes" >/root/.no_rootfs_resize`


If there is this file in the armbian system, it will be executed automatically when the system starts for the `first` time. It can be executed manually later: `armbian-fix`

**Comment by araczkowski** [comment, marked as answer]
to run the ``armbian-tf`` automatically after system starts (first time after shrinking the image) I added to the ``/etc/custom_service/start_service.sh``

``` bash
# Maximize root partition size
todo_rootfs_resize="/root/.no_rootfs_resize"
[[ -f "${todo_rootfs_resize}" && "$(cat ${todo_rootfs_resize} 2>/dev/null | xargs)" == "yes" ]] && {
    armbian-tf 2>/dev/null &&
    echo "[$(date +"%Y.%m.%d.%H:%M:%S")] Maximize root partition done!" >>${custom_log}
}
```

Now I'm doing:
1. run system, add packages and adjust settings
2. set the flag ``no_rootfs_resize`` to yes
3. shrinking the image in Gparted
4. burn the shrinked image to new USB/SD CARD
5. run the system from new image

and it works, after first boot, we have -> Maximize root partition size

![image](https://github.com/ophub/amlogic-s9xxx-armbian/assets/1263968/23dec31f-6c99-4338-9690-7318a235f8ec)

thanks @ophub for your help :+1: 
if you see better way to do this just please let me know

**Comment by ophub** [comment]
very good
